examples/generative/diffusion/training_loop.py

```diff
@@ -432,13 +450,19 @@ def training_loop(
             del data  # conserve memory
 
         # Save full dump of the training state.
-        if (
-            (state_dump_ticks is not None)
-            and (done or cur_tick % state_dump_ticks == 0)
-            and cur_tick != 0
-            and dist.rank == 0
-        ):
+        if dataset == "dfsr":
+            save_full_dump = (state_dump_ticks is not None) \
+                            and (done or cur_tick % state_dump_ticks == 0) \
+                            and dist.rank == 0
+        else:
+            save_full_dump = (state_dump_ticks is not None) \
+                            and (done or cur_tick % state_dump_ticks == 0) \
+                            and cur_tick != 0 \
```

Comment:
Let's just remove this and get rid of the entire conditional statement.

Reply:
Sure. The conditional statement has now been removed.